src/cli/cli.cpp

```diff
@@ -4981,6 +5043,24 @@ template <> otError Interpreter::Process<Cmd("service")>(Arg aArgs[])
 template <> otError Interpreter::Process<Cmd("netdata")>(Arg aArgs[]) { return mNetworkData.Process(aArgs); }
 
 #if OPENTHREAD_FTD
+/**
+ * @cli networkidtimeout (get,set)
+ * @code
+ * networkidtimeout 120
+ * Done
+ * @endcode
+ * @code
+ * networkidtimeout
+ * 120
+ * Done
+ * @endcode
+ * @cparam networkidtimeout [@ca{timeout}]
+ * Use the optional `timeout` argument to set the value of the `NETWORK_ID_TIMEOUT` parameter.
```

Comment:
Can we be more clear what the NETWORK_ID_PARAMETER is?  I think we need to update the API description for this... the get API has a notice about only using it for testing but the set one doesn't.  Seems like both should have the warning, and probably it should be applied here too.

Reply:
The `NETWORK_ID_TIMEOUT` is defined in Thread spec and is a constant (used in a bunch of places).

We have a warning for `otThreadGetNetworkIdTimeout()` and I agree it is more important to say this for `SetNetworkIdTimeout()` API:
```
* @note This API is reserved for testing and demo purposes only. Changing settings with
 * this API will render a production application non-compliant with the Thread Specification.
```

Since this API (and its CLI command) is intended for testing (mainly for folks who know the Thread spec and understand the impact of changing this), I think it may be good to add the warning but then we don't go into detail of what `NETWORK_ID_TIMEOUT` is and how it is used (keep it a mystery to avoid incorrect use).

Reply:
so, is the consensus that i will add the same warning that is currently in the Get API to both the Set API and the command ref?